plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestingH2JdbcClient.java

```diff
@@ -130,7 +132,15 @@ public boolean supportsAggregationPushdown(ConnectorSession session, JdbcTableHa
                 return Optional.of(timeColumnMapping(TIME_MILLIS));
 
             case Types.TIMESTAMP:
-                return Optional.of(timestampColumnMapping(TIMESTAMP_MILLIS));
+                TimestampType timestampType;
+                try {
+                    int decimalDigits = typeHandle.getRequiredDecimalDigits();
+                    timestampType = createTimestampType(decimalDigits);
+                }
+                catch (IllegalStateException ise) {
```

Comment:
if decimal digits may or may not be present, use optional-returning `typeHandle.getDecimalDigits()`

Reply:
addressed in https://github.com/trinodb/trino/pull/7591/commits/9d91cb558bb738b006a2d0d3b9070879f3b22c1a